slatedb/src/db/builder.rs

```diff
@@ -505,22 +505,18 @@ impl<P: Into<Path>> DbBuilder<P> {
         if inner.wal_enabled {
             inner.wal_buffer.init(task_executor.clone()).await?;
         };
-        task_executor
-            .add_handler(
-                MEMTABLE_FLUSHER_TASK_NAME.to_string(),
-                Box::new(MemtableFlusher::new(inner.clone(), manifest)),
-                memtable_flush_rx,
-                &tokio_handle,
-            )
-            .expect("failed to spawn memtable flusher task");
-        task_executor
-            .add_handler(
-                WRITE_BATCH_TASK_NAME.to_string(),
-                Box::new(WriteBatchEventHandler::new(inner.clone())),
-                write_rx,
-                &tokio_handle,
-            )
-            .expect("failed to spawn write batch event handler task");
+        task_executor.add_handler(
+            MEMTABLE_FLUSHER_TASK_NAME.to_string(),
+            Box::new(MemtableFlusher::new(inner.clone(), manifest)),
+            memtable_flush_rx,
+            &tokio_handle,
+        )?;
```

Comment:
Rather than panic, we should be returning the error. I did a check for expect's in this method and cleaned them up.